src/compat/reversediff.jl

```diff
@@ -0,0 +1,78 @@
+const RTR = ReverseDiff.TrackedReal
+const RTV = ReverseDiff.TrackedVector
+const RTM = ReverseDiff.TrackedMatrix
+using ReverseDiff: record_mul
+using ReverseDiff: SpecialInstruction
+
+_eps(::Type{<:RTR{T}}) where {T} = _eps(T)
+
+Base.:*(A::RTM{<:Any, D}, B::AbstractTriangular) where {D} = record_mul(A, B, D)
+Base.:*(A::AbstractTriangular, B::RTV{<:Any, D}) where {D} = record_mul(A, B, D)
+Base.:*(A::AbstractTriangular, B::RTM{<:Any, D}) where {D} = record_mul(A, B, D)
+function Base.:*(A::Adjoint{T, <:AbstractTriangular{T}}, B::RTM{<:Any, D}) where {T, D}
+    return record_mul(A, B, D)
+end
+function Base.:*(A::Adjoint{T, <:AbstractTriangular{T}}, B::RTV{<:Any, D}) where {T, D}
+    return record_mul(A, B, D)
+end
+
+function replace_diag(::typeof(log), X::RTM{<:Any, D}) where {D}
+    tp = ReverseDiff.tape(X)
+    X_value = ReverseDiff.value(X)
+    f(i, j) = i == j ? log(X_value[i, j]) : X_value[i, j]
+    out_value = f.(1:size(X_value, 1), (1:size(X_value, 2))')
+    function back(∇)
+        g(i, j) = i == j ? ∇[i, j]/X_value[i, j] : ∇[i, j]
+        return g.(1:size(X_value, 1), (1:size(X_value, 2))')
+    end
+    out = ReverseDiff.track(out_value, D, tp)
+    ReverseDiff.record!(tp, ReverseDiff.SpecialInstruction, replace_diag, (log, X), out, (back,))
+    return out
+end
+
+function replace_diag(::typeof(exp), X::RTM{<:Any, D}) where {D}
+    tp = ReverseDiff.tape(X)
+    X_value = ReverseDiff.value(X)
+    f(i, j) = i == j ? exp(X_value[i, j]) : X_value[i, j]
+    out_value = f.(1:size(X_value, 1), (1:size(X_value, 2))')
+    function back(∇)
+        g(i, j) = i == j ? ∇[i, j]*exp(X_value[i, j]) : ∇[i, j]
+        return g.(1:size(X_value, 1), (1:size(X_value, 2))')
+    end
+    out = ReverseDiff.track(out_value, D, tp)
+    ReverseDiff.record!(tp, ReverseDiff.SpecialInstruction, replace_diag, (exp, X), out, (back,))
+    return out
+end
+
+@noinline function ReverseDiff.special_reverse_exec!(instruction::SpecialInstruction{typeof(replace_diag)})
```

Comment:
Same here, just curious why these impls are neceesary

Reply:
This is part of the custom adjoint.